app/models/bulk_download.rb

```diff
@@ -293,10 +302,31 @@ def write_output_files_to_s3_tar_writer(s3_tar_writer)
                                     pipeline_runs.includes(sample: [:project])
                                   elsif download_type == BulkDownloadTypesHelper::CONTIG_SUMMARY_REPORT_BULK_DOWNLOAD_TYPE
                                     pipeline_runs.includes(:contigs, :sample)
+                                  elsif download_type == BulkDownloadTypesHelper::READS_NON_HOST_BULK_DOWNLOAD_TYPE
+                                    # We avoid pre-fetching "taxon_byteranges" here, even though it has to be fetched once per pipeline_run in the code below,
+                                    # because of the potentially large number of taxon_byteranges per pipeline_run.
+                                    pipeline_runs.includes(:sample)
                                   else
                                     pipeline_runs
                                   end
 
+    # Get the tax_level for the selected taxid by looking at the first available TaxonCount entry.
+    # Since TaxonCount isn't normalized, any TaxonCount entry would provide the same information.
+    if download_type == BulkDownloadTypesHelper::READS_NON_HOST_BULK_DOWNLOAD_TYPE
```

Comment:
for readability, I'd love to see all these boolean expressions converted to instance methods such as 

`isNonHostReadsType`

Reply:
I removed the `BulkDownloadTypesHelper::` by including that helper file in the model, which removes some of the wordiness. Instance methods for each download type feel a little overkill here.